testplan/runnable/interactive/base.py

```diff
@@ -736,30 +737,37 @@ def reload(self, rebuild_dependencies=False):
     def reload_report(self):
         """Update report with added/removed testcases"""
         new_report = self._initial_report()
-        for multitest in self.report.entries:  # multitest level
-            for suite_index, suite in enumerate(multitest.entries):
-                new_suite = new_report[multitest.uid][suite.uid]
-                for case_index, case in enumerate(suite.entries):
-                    try:
-                        if isinstance(case, TestGroupReport):
-                            for param_index, param_case in enumerate(
-                                case.entries
-                            ):
-                                try:
-                                    new_report[multitest.uid][suite.uid][
-                                        case.uid
-                                    ].entries[param_index] = case[
-                                        param_case.uid
-                                    ]
-                                except (KeyError, IndexError):
-                                    continue
-                        else:
-                            new_report[multitest.uid][suite.uid].entries[
-                                case_index
-                            ] = suite[case.uid]
-                    except (KeyError, IndexError):
-                        continue
-                multitest.entries[suite_index] = new_suite
+
+        def _preserve_partial_prev(prev, curr):
+            if isinstance(prev, TestGroupReport) and isinstance(
+                curr, TestGroupReport
+            ):
+                curr.timer = prev.timer
+                curr.status_override = prev.status_override
+                curr.env_status = prev.env_status
+                curr.logs = prev.logs
+
+                for uid in set(prev.entry_uids) & set(curr.entry_uids):
+                    _preserve_partial_prev(prev[uid], curr[uid])
+
+            elif isinstance(prev, TestCaseReport) and isinstance(
+                curr, TestCaseReport
+            ):
+                curr.timer = prev.timer
+                curr.runtime_status = prev.runtime_status
+                curr.status = prev.status
+                curr.attachments = prev.attachments
+                curr.status_override = prev.status_override
```

Comment:
Copying selected entries from prev to curr isn't very bulletproof, how do we know these are everything we need to copy? or what if we add new fields to report, we will forget to come back here to update.